client/src/components/Connections/components/NodeLoader.js

```diff
@@ -0,0 +1,15 @@
+// @flow
+import {withDataProviders} from 'data-provider'
+import {connectionEntityDetailProvider} from '../../../dataProviders/connectionsDataProviders'
+
+export type OwnProps = {
+  eid: string,
+}
+
+// Empty component used to preload node details shown by graph.
+// TODO: Ideas about a nicer solution?
+const NodeLoader = () => null
+
+export default withDataProviders((props: OwnProps) => [connectionEntityDetailProvider(props.eid)])(
+  NodeLoader
+)
```

Comment:
I'm not sure I understand why is this needed. Can't you wrap to `connectionEntityDetailProvider` the component, which is using this data? Or do you want to somehow preload it for performance reasons?

Reply:
The graph has some minimal entity info for each node. 
This component loads full entity detail for a single node/eid.
The idea is to display the graph (with lots of minimal nodes) asap and then load all the details in the background - showing more info in the graph as more details are being loaded.

I don't want to load all the details before the rendering of the graph - as it would take a long time to show it.
Can this be achieved in some better way? I'm open to suggestions.

Reply:
I see :thinking: Couldn't you use `needed` property of `data-provider`? See the example here: https://github.com/vacuumlabs/data-provider. I think if you set `needed` to `false`, you could render the component immediately and when you receive data, then update it.